src/mlpack/bindings/python/mlpack/arma_util.hpp

```diff
@@ -24,4 +24,124 @@ void SetMemState(T& t, int state)
   const_cast<arma::uhword&>(t.mem_state) = state;
 }
 
+/**
+ * Return the matrix's allocated memory pointer, unless the matrix is using its
+ * internal preallocated memory, in which case we copy that and return a
+ * pointer to the memory we just made.
+ */
+inline double* GetMemory(arma::mat& m)
```

Comment:
Couldn't you write something like:

```c++
template<typename MatType>
inline typename MatType::elem_type* GetMemory(MatType& m)
{
}
```

I was thinking if we are going for: https://github.com/mlpack/mlpack/blob/master/doc/policies/elemtype.hpp I would have to add a bunch more, perhaps I missed something.

Reply:
I was afraid Cython would have problems with the `MatType::elem_type` bit but I never tried it---let me do that now.